### PR TITLE
feat(rest-assured): add support for headers blacklist

### PR DESCRIPTION
**Pull request by antonfilichkin**
### Context
RestAssured from version 4.2.0 added possibility to blacklist headers in the request or response logs.
~~This adds a configurable option to consider blacklisted headers from the RestAssured LogConfiguration and blacklist them in the Allure report as well. Enabled by default, so if there is a need to expose header blacklisted by RestAssured in Allure report it must be set explicitly.~~
This makes Allure report to blacklist the same headers.

Example of header blacklisted in Allure report:
image

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: allure-rest-assured/src/main/java/io/qameta/allure/restassured/AllureRestAssured.java [hunk 1]
```diff
@@ -65,6 +72,16 @@ public AllureRestAssured setResponseAttachmentName(final String responseAttachme
         return this;
     }
 
+    /**
+     * Configure filter to consider headers that are blacklisted by RestAssured LogConfig. Enabled by default.
+     * @see io.restassured.config.LogConfig#blacklistHeader
+     * @see io.restassured.config.LogConfig#blacklistHeaders
+     */
+    public AllureRestAssured followHeadersBlacklist(final boolean isEnabled) {
```
Comment (baev) [inline allure-rest-assured/src/main/java/io/qameta/allure/restassured/AllureRestAssured.java:80]: Do we need to have the possibility to override the default behaviour and expose blacklisted headers in the report?

Comment (antonfilichkin) [reply allure-rest-assured/src/main/java/io/qameta/allure/restassured/AllureRestAssured.java:80]: They are exposed now. But from my perspective - it doesn't make any sense to blacklist them in logs but expose them in Allure report.
Should I remove the configuration option and make blacklist-follow the only behavior?

Comment (baev) [reply allure-rest-assured/src/main/java/io/qameta/allure/restassured/AllureRestAssured.java:80]: > Should I remove the configuration option and make blacklist-follow the only behavior?

Yes, let's start with that. We can introduce the configuration option later if requested by the community

**Comment by baev** [comment]
Meanwhile, I don't like the used wording (`blacklist` and `blacklisted`). Now, most software uses `allowlist` and `denylist` instead. However, this is the [exact copy of the behaviour from RestAssured](https://github.com/rest-assured/rest-assured/blob/034ebfb0db4a6302e3bd429d26a2eb79e3190e9e/changelog.txt#L304), and I don't see any point to introduce additional UX complications for RestAssured users.

File: allure-rest-assured/src/main/java/io/qameta/allure/restassured/AllureRestAssured.java (hunk 1 above)
Comment (antonfilichkin) [reply allure-rest-assured/src/main/java/io/qameta/allure/restassured/AllureRestAssured.java]: Removed configurable option - always reads log config.

**Review by baev — APPROVED** [review]

**Comment by baev** [comment]
@antonfilichkin thanks!

**Pull request merged**